py-polars/polars/internals/expr.py

```diff
@@ -1675,7 +1661,9 @@ def rank(self, method: str = "average") -> "Expr":  # type: ignore
         """
         return wrap_expr(self._pyexpr.rank(method))
 
-    def diff(self, n: int = 1, null_behavior: str = "ignore") -> "Expr":  # type: ignore
+    def diff(
+        self, n: int = 1, null_behavior: Literal["ignore", "drop"] = "ignore"
```

Comment:
`null_behavior` should alosohave `str` as type as with `Literal` only .

```python
my_variable = "ignore"

diff (..., null_behavior=my_variable)
```
As using a variable with that string inside, will not typecheck.

Not sure if we better use an Union or an overload
https://www.python.org/dev/peps/pep-0586/#interactions-with-overloads

Reply:
That is an important gotcha. I was under the impression that would pass, and that therefore `Literal` would help us to tighten up the types by changing all string options into Literals...